src/Lower.cpp

```diff
@@ -808,7 +808,21 @@ class FindBuffers : public IRGraphVisitor {
     using IRGraphVisitor::visit;
 
     void visit(const Call *op) {
-        IRGraphVisitor::visit(op);
+
+        if (op->call_type == Call::Extern) {
```

Comment:
The variable ends up with an associated Parameter object:

In the test use case we are creating a variable named "i0" + ".buffer", in order to pass the image parameter i0 to an extern call.  Creating an ordinary variable with this name asserts in the test code expression, inside the CheckVars visitor of an assignment operator, because the variable "i0.buffer" is undefined (Function.cpp:95).

We can make the new variable instance a Parameter type variable (the rationale being that the image that we are attempting to access is a in fact parameter), but next we crash (or assert with the fix below) in Lower.cpp add_image_checks. The boxes_touched visitor ignores extern calls and there is no information about the bounds of this image.

This check in FindBuffers avoids tracking buffers in add_image_checks in the narrow case that they are only passed directly to extern calls.